dataprofiler/data_readers/data_utils.py

```diff
@@ -843,3 +846,138 @@ def url_to_bytes(url_as_string: Url, options: Dict) -> BytesIO:
+        # Check if the path starts with any of the specified prefixes
+        is_s3 = any(path.startswith(prefix) for prefix in s3_uri_prefixes)
+
+        if not is_s3:
+            logger.info(f"'{path}' is not a valid S3 URI")
+
+        return is_s3
+
+    @staticmethod
+    def _create_boto3_client(
+        aws_access_key_id: Optional[str],
+        aws_secret_access_key: Optional[str],
+        aws_session_token: Optional[str],
+        region_name: Optional[str],
+    ) -> boto3.client:
+        return boto3.client(
+            "s3",
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+            region_name=region_name,
+        )
+
+    @staticmethod
+    def create_s3_client(
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        region_name: Optional[str] = None,
+    ) -> boto3.client:
+        """
+        Create and return an S3 client.
+
+        Args:
+            aws_access_key_id (str): The AWS access key ID.
+            aws_secret_access_key (str): The AWS secret access key.
+            aws_session_token (str): The AWS session token
+                (optional, typically used for temporary credentials).
+            region_name (str): The AWS region name (default is 'us-east-1').
+
+        Returns:
+            boto3.client: A S3 client instance.
+        """
+        # Check if credentials are not provided
+        # and use environment variables as fallback
+        if aws_access_key_id is None:
+            aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID")
+        if aws_secret_access_key is None:
+            aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY")
+        if aws_session_token is None:
+            aws_session_token = os.environ.get("AWS_SESSION_TOKEN")
+
+        # Check if region is not provided and use environment variable as fallback
+        if region_name is None:
+            region_name = os.environ.get("AWS_REGION", "us-east-1")
+
+        # Check if IAM roles for service accounts are available
+        try:
+            s3 = S3Helper._create_boto3_client(
+                aws_access_key_id, aws_secret_access_key, aws_session_token, region_name
+            )
+        except botocore.exceptions.NoCredentialsError:
+            # IAM roles are not available, so fall back to provided credentials
+            if aws_access_key_id is None or aws_secret_access_key is None:
+                raise ValueError(
+                    "AWS access key ID and secret access key are required."
+                )
+            s3 = S3Helper._create_boto3_client(
+                aws_access_key_id, aws_secret_access_key, aws_session_token, region_name
+            )
+
+        return s3
+
+    @staticmethod
+    def s3_uri_to_bytes(s3_uri: str, s3_client: boto3.client) -> BytesIO:
```

Comment:
```suggestion
    def get_s3_uri_to_bytes(s3_uri: str, s3_client: boto3.client) -> BytesIO:
```

Reply:
refactoring the function name to `get_s3_uri`.